homeassistant/components/shelly/config_flow.py

```diff
@@ -397,13 +397,13 @@ async def _async_get_info(self, host: str, port: int) -> dict[str, Any]:
 
     @staticmethod
     @callback
-    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlowHandler:
+    def async_get_options_flow(config_entry: ShellyConfigEntry) -> OptionsFlowHandler:
```

Comment:
You generally don't need to use the custom `ConfigEntry` type inside the ConfigFlow / OptionsFlow as you don't access the `runtime_data` there. I would just revert these here.